lib/mayaHydra/flowViewportAPIExamples/footPrintNode/mhFootPrintNode.cpp

```diff
@@ -0,0 +1,666 @@
+            {
+                plug.setValue(dummyInputVal + 1);//Dirty one parameter that affects the mDummyOutput attribute
+                MPlug plugOutput(thisMObject(), mDummyOutput);
+                if (!plugOutput.isNull())
+                {
+                    int dummyOutputVal;
+                    plugOutput.getValue(dummyOutputVal);//This will trigger a call to compute
+                }
+            }
+        }
+    }
+}
+
+void MhFootPrint::_UpdateThisMObject()
+{
+    if (! _thisMObject.isNull()){
+        return;
+    }
+
+    MObject currentMObj = thisMObject();
+    if (_oldMObject.isNull() || _oldMObject != currentMObj){
+        _thisMObject = currentMObj;     
+        _oldMObject  = _thisMObject;
+    }
+}
+
+void MhFootPrint::setupFlowViewportInterface()
+{
+    static const SdfPath noPrefix = SdfPath::AbsoluteRootPath();
+
+    _UpdateThisMObject();
+
+    //Remove the callback
+    if (_cbAttributeChangedId){
+        CHECK_MSTATUS(MMessage::removeCallback(_cbAttributeChangedId));
+        _cbAttributeChangedId = 0;
+    }
+    //Add the callback when an attribute of this node changes
+    _cbAttributeChangedId = MNodeMessage::addAttributeChangedCallback(_thisMObject, attributeChangedCallback, ((void*)this));
+
+    //Remove the previous data producer scene index in case it was registered, if that occurs it's because _thisMObject has changed and we want to update the maya callbacks on the node
+    Fvp::DataProducerSceneIndexInterface& dataProducerSceneIndexInterface = Fvp::DataProducerSceneIndexInterface::get();
+    dataProducerSceneIndexInterface.removeViewportDataProducerSceneIndex(_retainedSceneIndex, pxr::FvpViewportAPITokens->allViewports);
+
+    //Data producer scene index interface is used to add the retained scene index to all viewports with all render delegates
+    dataProducerSceneIndexInterface.addDataProducerSceneIndex(_retainedSceneIndex, noPrefix, (void*)&_thisMObject, FvpViewportAPITokens->allViewports,FvpViewportAPITokens->allRenderers);
+}
+
+// Retrieve value of the size attribute from the node
+float MhFootPrint::_GetSizeInCentimeters() const
+{
+    const MObject obj = thisMObject();
+    
+    MPlug plug(obj, MhFootPrint::mSize);
+    if (!plug.isNull())
+    {
+        MDistance sizeVal;
+        if (plug.getValue(sizeVal))
+        {
+            return (float)sizeVal.asCentimeters();
+        }
+    }
+
+    return 1.0f;
+}
+
+// Retrieve value of the color attribute from the node
+GfVec3f MhFootPrint::_GetColor() const
+{
+    const MObject obj = thisMObject();
+    MPlug plug(obj, MhFootPrint::mColor);
+    if (!plug.isNull())
+    {
+        double3 color;
+        GetDouble3AttributeValue(color, obj, MhFootPrint::mColor);
+        return {(float)color[0], (float)color[1], (float)color[2]};
+    }
+
+    return GfVec3f(0.f,0.f,1.f);
+}
+
+MStatus MhFootPrint::compute( const MPlug& plug, MDataBlock& dataBlock)
+{
+    if( ! _init){
+        _CreateAndAddFootPrintPrimitives();
```

Comment:
It seems strange to me that we need to use the Dataflow Graph compute method to register Hydra drawing.  Why can't this be done on node creation?

Reply:
Can we move this code to node creation time?  How about using a path?  There does not need to be a node if we use a Ufe::Path, and an MDagPath can be built up.

Reply:
Let's take this offline, I tried unsuccessfully so far.